Hotfix for is_int()

TheKeymaster opened the pull request:
We are checking for `is_int()`, but the _GET param always returns a string, wich results in an error when setting a valid param like `start=0`.

Commit:
Hotfix for superglobal _GET always returns a string, but we are checking for is_int, wich not allows strings.

TheKeymaster commented:
@georgms could you review this, so I can merge the hotfix? I have

 * fixed this issue.
 * added a test for it.

georgms commented on index.php:
```diff
@@ -8,8 +8,8 @@
 // Get URL params
-$start = $outputXmlHelper->getUrlParam($outputXmlHelper::START_NAME, $outputXmlHelper::DEFAULT_START_VALUE, $_GET);
-$count = $outputXmlHelper->getUrlParam($outputXmlHelper::COUNT_NAME, $outputXmlHelper::DEFAULT_COUNT_VALUE, $_GET);
+$start = (int)$outputXmlHelper->getUrlParam($outputXmlHelper::START_NAME, $outputXmlHelper::DEFAULT_START_VALUE, $_GET);
```
You could make `getUrlParam` type safe by using the `filter_var` method and adding a fourth argument to `getUrlParam` with the desired type, eg. `FILTER_VALIDATE_INT` or `FILTER_VALIDATE_FLOAT`.

georgms reviewed (approved):
`getUrlParam` could be type-safe. I leave it up to you to implement this.

Commit:
Fix review issues and merged some functions together

TheKeymaster commented:
@georgms could you review again?

georgms commented on OutputXMLHelper.php:
```diff
@@ -91,4 +95,23 @@ public function getXml($start, $count)
+     * Sets the export error header and prints an error message
+     */
+    public function throwError() {
+        header(self::EXPORT_ERROR_HEADER, true, self::EXPORT_ERROR_CODE);
```
Isn't there something like this already built in Dokuwiki?

georgms commented on index.php:
```diff
@@ -8,13 +8,11 @@
 // Get URL params
-$start = $outputXmlHelper->getUrlParam($outputXmlHelper::START_NAME, $outputXmlHelper::DEFAULT_START_VALUE, $_GET);
-$count = $outputXmlHelper->getUrlParam($outputXmlHelper::COUNT_NAME, $outputXmlHelper::DEFAULT_COUNT_VALUE, $_GET);
+$start = (int)$outputXmlHelper->getUrlParam($outputXmlHelper::START_NAME, $outputXmlHelper::DEFAULT_START_VALUE, $_GET, FILTER_VALIDATE_INT);
```
Casting to `int` shouldn't be necessary anymore, now that the type is checked.

TheKeymaster replied:
It still is needed, because the _GET param will always return a string. So I need to manually convert the string, containing the integer value, to an integer.

`count=2` → `$_GET['count'] = '2'`.

georgms replied:
`filter_var` does not only validate the type but also converts it if possible. So `getUrlParam` should run the value through `filter_var` not only for validation but also for converting the type.

georgms reviewed (changes requested):
Nice job! Just two minor questions.

Commit:
Fix review issue valudation

Commit:
Fix review issue, now use DokuWiki function for error printing

Commit:
Using a const for error level

TheKeymaster commented:
@georgms I have made some changes;  
 * First up I am validating the params in the function `\OutputXMLHelper::paramsValid` now, because this function was already made to validate the params and therefor should also include the integer validation.   

 * Secondly DokuWiki is using their own global functions for printing out messages. Usually messages will be sent via the `msg()` function, which generates an array of all messages (You can also include the error_level e.g. information, warning or error). DokuWiki will print all messages with the function `html_msgarea()`, which generates a `<div>` container with classes based on the error level, around foreach array value, that was sent using the `msg()` function. 

@georgms could you review those changes?

georgms commented on _test/outputxmlhelper.test.php:
```diff
@@ -66,21 +66,22 @@ public function parameterProviderForXMLCallWithInvalidParams()
-    function test_export_call_works_when_calling_export_with_float_params($start, $count) {
+    function test_export_call_works_when_calling_export_with_float_or_string_params($start, $count) {
         $outputXmlHelper = new OutputXMLHelper();
         $this->assertEquals(false, $outputXmlHelper->paramsValid($start, $count), 'Expected float params should be invalid.');
```
Missing `$type` argument

georgms reviewed (changes requested):
`paramsValid` is called without the `$type` parameter in the tests. Please fix and check why the test did not fail.

Commit:
This commit should fail

Commit:
Yet another try. Please fail

Commit:
Possible fix

Commit:
Add group settings for phpunit in order to execute those tests

TheKeymaster commented:
@georgms gawd... I actually forgot that group parameter and those tests were never executed. The reason for this was that in the `.travis.yml` I explicitly set the parameter `--group plugin_findologicxmlexport`. Now I've added the comment block where i set this group also to the file `_test/outputxmlhelper.test.php`. Could you review this again?

georgms reviewed (changes requested):
I could not see a failing build in Travis. Are you sure the tests are running properly?

TheKeymaster commented:
> Are you sure the tests are running properly?

I am sure. I already fixed those tests and before there were 

```
OK (23 tests, 68 assertions)
```

and now there are

```
OK (38 tests, 83 assertions)
```

tests running.  
I double checked all other tests and it seems like everything looks fine.  
*As a side note I fixed the tests in https://github.com/findologic/dokuwiki-plugin-findologic-xml-export/pull/39/commits/42b6deeade1ac35d9027b309514e5bcfa2b8723c.*

I would be glad to merge this :).

georgms commented:
@TheKeymaster Increasing the number of tests is good, but I want to see the test fail (without the fix).

Commit:
This commit should fail with 15 failing tests

TheKeymaster commented:
@georgms the build failed with

```
Tests: 38, Assertions: 68, Errors: 15.
```

I will fix them now again ;)

Commit:
Fix 15 broken tests
